### PR TITLE
add node-exporter to allowed ports

### PR DESCRIPTION
**Pull request by zetaab**
The problem what we currently have in openstack is that node-exporter does not work correctly. The default behaviour in [kube-prometheus](https://github.com/coreos/kube-prometheus) is that it will expose node-exporter using hostnetwork https://github.com/coreos/kube-prometheus/blob/master/manifests/node-exporter-daemonset.yaml#L68 this means that port 9100 should be open to prometheus itself (running in normal nodes). 

Screenshot 2019-05-13 at 13 00 06

After I opened 9100 port ingress in both master+node SG it works

/sig openstack
/kind feature

### CODE REVIEW

File: pkg/model/openstackmodel/firewall.go [hunk 1]
```diff
@@ -294,6 +294,26 @@ func (b *FirewallModelBuilder) addKubeletRules(c *fi.ModelBuilderContext, sgMap
...
+	nodeExporterIngress := &openstacktasks.SecurityGroupRule{
+		Lifecycle:    b.Lifecycle,
+		Direction:    s(string(rules.DirIngress)),
+		Protocol:     s(IPProtocolTCP),
+		EtherType:    s(IPV4),
+		PortRangeMin: i(9100),
+		PortRangeMax: i(9100),
+	}
+	// allow 9100 port from nodeSG
+	addDirectionalGroupRule(c, masterSG, nodeSG, nodeExporterIngress)
```
Comment (zetaab) [inline pkg/model/openstackmodel/firewall.go]: @drekle is this correct? so I want that 9100 port in master security group is open when connecting from nodeSG

Comment (drekle) [reply pkg/model/openstackmodel/firewall.go]: Yeah this is correctly interpreted.  Perhaps if you can find better wording go for it.

File: pkg/model/openstackmodel/firewall.go [hunk 2]
```diff
@@ -294,6 +294,26 @@ func (b *FirewallModelBuilder) addKubeletRules(c *fi.ModelBuilderContext, sgMap
 	return nil
 }
 
+// addNodeExporterRules - Allow 9100 TCP port from nodesg
```
Comment (drekle) [inline pkg/model/openstackmodel/firewall.go]: I don't believe that Kops can be responsible for deploying prometheus, or prometheus-operator.  Kops cant be responsible for the security groups of applications it's not also responsible for managing.

I think we can use additionalSecurityGroups as defined in the instance group spec to enable this, though it seems more manual.  Additionally we might provide a way to declare in the spec that you need additional security groups created.

Comment (zetaab) [reply pkg/model/openstackmodel/firewall.go]: this same deployment (kube-prometheus) works fine in aws kops clusters. @chrisz100 @mikesplain do you have idea should we allow this or not? Should openstack behave in similar way than aws?

Comment (drekle) [reply pkg/model/openstackmodel/firewall.go]: I actually need this too.  Still interested in thoughts from @chrisz100  and @mikesplain, however I think I will approve this for now and come up with any other solutions I feel might be more appropriate as needed.

We still have the flexibility of changing things as this is still labeled alpha.

/lgtm

**Pull request merged**